src/main/java/bio/terra/cli/serialization/userfacing/resource/UFBqDataset.java

```diff
@@ -21,15 +22,18 @@
   public final String projectId;
   public final String datasetId;
   public final String location;
+  public final Integer numTables;
 
   /** Serialize an instance of the internal class to the command format. */
   public UFBqDataset(BqDataset internalObj) {
     super(internalObj);
     this.projectId = internalObj.getProjectId();
     this.datasetId = internalObj.getDatasetId();
 
-    Optional<Dataset> dataset = internalObj.getDataset();
+    GoogleBigQuery bigQuery = GoogleBigQuery.fromContextForPetSa();
+    Optional<Dataset> dataset = bigQuery.getDataset(projectId, datasetId);
     this.location = dataset.isPresent() ? dataset.get().getLocation() : null;
```

Comment:
nit: you can use map/orElse instead of the ternary operator.

Reply:
done.